src/highlighter.js

```diff
@@ -517,6 +642,7 @@ export default function (options) {
      * @param {Object[]} highlightIndex
      */
     function buildHighlightIndex(rootNode, highlightIndex) {
+        //TODO: if needed, fix according to changes
```

Comment:
This might depend on saving & restoring highlights (TRN-894), right? Do you think we need to check that before finalising this PR?

Reply:
We agreed last week to not merge this before TRN-895 is done.
Sad that it delays testing, but you can see in github actions that pipeline can't build tao-deliver-fe with git-branch dependency

Reply:
I pushed changes to the same PR both here and in `tao-deliver-fe` - because in TRN-894 I added a 'model', so implementation of TRN-967 in `tao-deliver-fe` also changed a lot